pylint/checkers/typecheck.py

```diff
@@ -523,6 +525,77 @@ def _emit_no_member(
     return True
 
 
+def _get_all_attribute_assignments(
+    node: astroid.FunctionDef, name: str | None = None
+) -> set[str]:
+    attributes: set[str] = set()
+    for child in node.nodes_of_class((astroid.Assign, astroid.AnnAssign)):
+        targets = []
+        if isinstance(child, astroid.Assign):
+            targets = child.targets
+        elif isinstance(child, astroid.AnnAssign):
+            targets = [child.target]
+        for assign_target in targets:
+            if isinstance(assign_target, astroid.Tuple):
+                targets.extend(assign_target.elts)
+                continue
+            if (
+                isinstance(assign_target, astroid.AssignAttr)
+                and isinstance(assign_target.expr, astroid.Name)
+                and (name is None or assign_target.expr.name == name)
+            ):
+                attributes.add(assign_target.attrname)
+    return attributes
+
+
+def _enum_has_attribute(
+    owner: astroid.Instance | nodes.ClassDef, node: nodes.Attribute
+) -> bool:
+    if isinstance(owner, astroid.Instance):
+        enum_def = next(
+            (b.parent for b in owner.bases if isinstance(b.parent, nodes.ClassDef)),
+            None,
+        )
+        if enum_def is None:
+            return False
```

Comment:
Let's do:
```suggestion
        assert enum_def
```

I think this will never be `None` right?